### PR TITLE
chore(test): Tidy up python tool

### PR DESCRIPTION
**Pull request by Callisto13**
**What this PR does / why we need it**:

Only the second commit does anything new, the rest is tidying up.

The main purpose of this PR is to fix the e2e nightly build. I think I must
have messed up a rebase or something because I had run the exact
command so many times and it was fine so 🤷‍♀️ .

I also did a bit of tidying up in the `test/tools/` dir. i have no idea how
a python project should be structured (even a toy one like this) but it was a
bit messy in there and there is much room for refactoring and tests so
organising things better now felt sensible.

### CODE REVIEW

File: test/tools/config/config.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 class Config:
     def __init__(self):
         self.dir = dirname(abspath(__file__))
-        self.base = dirname(dirname(dirname(abspath(__file__))))
+        self.base = dirname(dirname(dirname(dirname(abspath(__file__)))))
```
Comment (yitsushi) [inline test/tools/config/config.py:11]: Maybe with `pathlib.Path` it's more readable:
```
from pathlib import Path
```

```suggestion
        self.base = str(Path(abspath(__file__)).parent.parent.parent.parent)
```

**Review by yitsushi — APPROVED** [review]
Small style suggestion, not a blocker. lgtm

File: test/tools/config/config.py (hunk 1 above)
Comment (yitsushi) [reply test/tools/config/config.py:11]: In theory, `parents[4]` should work too, but then we have to check the length first

Comment (Callisto13) [reply test/tools/config/config.py:11]: > Maybe with `pathlib.Path` it's more readable

maybe, i will add that with my next pr in all the places we do this 👍

**Pull request merged**